transports/dns/src/lib.rs

```diff
@@ -49,16 +49,30 @@
 //!      problematic on platforms like Android, where there's a lot of
 //!      complexity hidden behind the system APIs.
 //! If the implementation requires different characteristics, one should
-//! consider providing their own implementation of [`GenDnsConfig`] or use
+//! consider providing their own implementation of [`Config`] or use
 //! platform specific APIs to extract the host's DNS configuration (if possible)
 //! and provide a custom [`ResolverConfig`].
 //!
 //![trust-dns-resolver]: https://docs.rs/trust-dns-resolver/latest/trust_dns_resolver/#dns-over-tls-and-dns-over-https
 
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 
+mod config;
```

Comment:
Do we necessarily need this module? I think it would be slightly easier to maintain if we would just make two inline modules:

```rust
#[cfg(feature = "tokio")]
pub mod tokio {

}

#[cfg(feature = "async-std")]
pub mod async_std {

}
```